File: PlusOne/plusOne.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        # По условию если в массиве один элемент и он не равен 9,то:
        if len(digits) == 1 and digits[0] == 9:
            # Возвращаю 1 и 0 по условию
            return [1, 0]
        # Если же последняя цифра массиве не 9, то присваиваю ей ее значение + 1
        if digits[-1] != 9:
            digits[-1] += 1
            # Возвращаю массив по условию
            return digits
        # Либо
        else:
            # Делаю последнюю цифру 0
            digits[-1] = 0
            # Прохожусь рекурсией
            digits[:-1] = self.plusOne(digits[:-1])
            # Возвращаю массив по условию
            return digits  


# ! Решение в одну строку с мапой
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        return map(int, list(str(int(''.join(map(str, digits))) + 1)))

File: PlusOne/plusOne.js
var plusOne = function(digits) {
    // Прохожусь циклом по массиву
    for (let i = digits.length - 1; i >= 0; i--) {
        // Условие, если число в массиве больше 9, ++ - играет роль обычного инкремента(+1)
        if (++digits[i] > 9) {
            // То по условию, делаю число нулем
            digits[i] = 0
        } else {
            // Либо возвращаю массив
            return digits
        }
    }
    // По условию приходится добавлять в начало массива 1, чтобы при инпуте с цифрой 9 вышел такой массив: [1, 0]
    digits.unshift(1);
    // Возвращаю сам массив
    return digits;
};

// ! Тупо синтакс сахар без скобок, тоже самое что сверху
// var plusOne = function(digits) {
//     for(let i = digits.length - 1; i >= 0; i--){
//     if(++digits[i] > 9) digits[i] = 0;
//     else return digits;
//   }
//     digits.unshift(1)
//     return digits
// };